### ISSUE TITLE
[BUG]: <Fix for Dev Proxy Installation Script on Linux: Addressing chmod Error and xattr Compatibility Issue>

### ISSUE THREAD

**Issue by SilentSobs** [microsoft/dev-proxy#558]
### Description

The Dev Proxy installation script encounters two issues when executed on a Linux system:

Chmod Issue:

Issue: The script attempts to configure devproxy and its files as executable using the chmod command but encounters an error: chmod: cannot access './libe_sqlite3.dylib': No such file or directory.
Expected Behavior: The script should use the correct file name (libe_sqlite3.so) in the chmod command.
Proposed Fix: Update the chmod command to chmod +x ./devproxy ./libe_sqlite3.so.

xattr Issue:

Issue: The script tries to remove the quarantine attribute using the xattr command, which is specific to macOS and not available on Linux.
Expected Behavior: The script should gracefully handle the absence of the xattr command on Linux and print a warning message before skipping the quarantine attribute removal.
Proposed Fix: Check for the existence of the xattr command before attempting to use it. If not found, print a warning and skip the quarantine attribute removal.

### Expected behaviour

After running the Dev Proxy installation script on Linux using the provided command:
```
bash -c "$(curl -sL https://aka.ms/devproxy/setup.sh)"

```
I expect the script to correctly configure devproxy and its files as executable without encountering the chmod: cannot access './libe_sqlite3.dylib': No such file or directory error.
---------------------------------------------------------------------------------------------
Regarding xattr Issue:
Expected Behavior:
When running the script on a Linux system, I expect the script to intelligently handle the absence of the xattr command. Specifically, the script should print a warning message stating that the xattr command is not found and gracefully skip the quarantine attribute removal step.

### Actual behaviour

Regarding chmod Issue:
Actual Behavior:
Upon running the Dev Proxy installation script on Linux with the provided command:
```
bash -c "$(curl -sL https://aka.ms/devproxy/setup.sh)"

```
The script encounters an error during the configuration of devproxy and its files as executable. The specific error message is:
```
chmod: cannot access './libe_sqlite3.dylib': No such file or directory

```

Regarding xattr Issue:
Actual Behavior:
When running the script on a Linux system, the script attempts to use the xattr command to remove the quarantine attribute. However, the xattr command is specific to macOS and is not available on Linux. As a result, the script fails during this step.

### Steps to reproduce

```
bash -c "$(curl -sL https://aka.ms/devproxy/setup.sh)"

```

### Dev Proxy Version

-

### Operating system (environment)

Linux

### Shell

bash

### Configuration file

```
bash -c "$(curl -sL https://aka.ms/devproxy/setup.sh)"

```

### Additional Info

1) Fix for chmod Issue:

Upon running the Dev Proxy installation script on Linux, the script encounters an error during 
the configuration of devproxy and its files as executable. The specific error message is:

```
chmod: cannot access './libe_sqlite3.dylib': No such file or directory

```

Proposed Fix:
Update the chmod command to use the correct file name:

```
chmod +x ./devproxy ./libe_sqlite3.so

```

2) Fix for xattr Issue:

Proposed Fix:
Check if the xattr command exists before attempting to use it. If the command is not found, print a warning and skip the quarantine attribute removal:

```
# Check if xattr command exists
if command -v xattr &> /dev/null; then
    echo "Removing quarantine attribute from devproxy and its files..."
    xattr -d com.apple.quarantine ./devproxy ./libe_sqlite3.so
else
    echo "Warning: xattr command not found. Skipping quarantine attribute removal."
fi

```

**Comment by waldekmastykarz** [comment]
Good catch @SilentSobs and thanks for bringing it up. `libe_sqlite3.dylib` is the name of the library on macOS, so we need to update the script to use that name on macOS and `libe_sqlite3.so` on linux.

**Comment by SilentSobs** [comment]
I have tested this on Linux but need to check it on macOS. I don't have a macOS environment, so please confirm the following script:

```
rm ./devproxy.zip
echo "Configuring devproxy and its files as executable..."
if [[ "$OSTYPE" == "darwin"* ]]; then
    LIBRARY_NAME="libe_sqlite3.dylib"
    echo "Configuring devproxy and its files as executable..."
    chmod +x ./devproxy ./"$LIBRARY_NAME"

    echo "Removing quarantine attribute from devproxy and its files..."
    xattr -d com.apple.quarantine ./devproxy ./"$LIBRARY_NAME"
else
    # Assume Linux
    LIBRARY_NAME="libe_sqlite3.so"
    echo "Configuring devproxy and its files as executable..."
    chmod +x ./devproxy ./"$LIBRARY_NAME"
fi
```

**Comment by waldekmastykarz** [comment]
I suggest that we avoid repeating the `echo "Configuring devproxy and its files as executable..."` message inside the `if` clause. The only thing that we need to wrap in `if` is setting the library name. Everything else should be outside. Also, let's make the library name variable lowercase to clearly indicate that it's a local variable rather than global.

Shouldn't btw `./"$library_name"` be `"./$library_name"`? It seems odd to add quotes mid-path.

**Comment by waldekmastykarz** [comment]
Btw, we removed xattr from the setup script, because it turns out that we didn't need it after all

**Comment by SilentSobs** [comment]
Ohk. Got It.

**Comment by SilentSobs** [comment]
```
rm ./devproxy.zip

# Set library name based on the OS type
if [[ "$OSTYPE" == "darwin"* ]]; then
    library_name="libe_sqlite3.dylib"
else
    # Assume Linux
    library_name="libe_sqlite3.so"
fi

echo "Configuring devproxy and its files as executable..."
chmod +x ./devproxy "./$library_name"

```

**Comment by SilentSobs** [comment]
I noticed you reacted with a thumbs-up to the code  there any further changes is it fine ?

**Comment by waldekmastykarz** [comment]
The code above looks good. No further comments

**Comment by waldekmastykarz** [comment]
Hey @SilentSobs, are you still planning to submit a PR?

**Comment by SilentSobs** [comment]
Oh Sorry Yes I would like to submit

**Comment by SilentSobs** [comment]
Sorry for delay i will make a pull request today itself
